internal/iam/edit.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Mia srl
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iam
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mia-platform/miactl/internal/client"
+	"github.com/mia-platform/miactl/internal/resources"
+)
+
+func EditUserRole(ctx context.Context, client *client.APIClient, companyID, resourceID string, change resources.EditIAMRole) (*client.Response, error) {
+	return EditIAMResourceRole(ctx, client, companyID, resourceID, UsersEntityName, change)
+}
+
+func EditGroupRole(ctx context.Context, client *client.APIClient, companyID, resourceID string, change resources.EditIAMRole) (*client.Response, error) {
+	return EditIAMResourceRole(ctx, client, companyID, resourceID, GroupsEntityName, change)
+}
+
+func EditServiceAccountRole(ctx context.Context, client *client.APIClient, companyID, resourceID string, change resources.EditIAMRole) (*client.Response, error) {
+	return EditIAMResourceRole(ctx, client, companyID, resourceID, ServiceAccountsEntityName, change)
+}
+
+func EditIAMResourceRole(ctx context.Context, client *client.APIClient, companyID, resourceID, entityType string, change resources.EditIAMRole) (*client.Response, error) {
```

Comment:
Is this function public, or used only by the above functions?
In that case, instead the switch case it could be passed directly the apiPath. Otherwise, why there are the above functions?

Reply:
Yeah it was private during the initial concept of the module, but then I've made it public for easier use in the company command. I've deleted the three specific one and left only the last one to simplify things.